autoprotocol/builders.py

```diff
@@ -1557,38 +1576,127 @@ def device_mode_params(
-            raise ValueError(f"Tempest chip model is {model}. It must be: high_volume.")
-        if (nozzle != "standard") and (nozzle is not None):
-            raise ValueError(f"Tempest nozzle is {nozzle}. It must be: standard.")
-        if (chip_material not in ["silicone", "pfe"]) and (chip_material is not None):
-            raise ValueError(
-                f"Tempest chip material is {chip_material}. It must be either: silicone or pfe."
-            )
-
-        device_mode_params = {}
-        if any([model, chip_material, nozzle]):
-            if model is None:
-                model = "high_volume"
-            if chip_material is None:
-                chip_material = "pfe"
-            if nozzle is None:
-                nozzle = "standard"
-            x_tempest_chip = {
+            If device is not of the following: [x_tempest_chip, x_mantis]
+        """
+        params: OrderedDict = OrderedDict(
+            {
                 "model": model,
                 "material": chip_material,
                 "nozzle": nozzle,
+                "diaphragm": diaphragm,
+                "nozzle_size": nozzle_size,
+                "tubing": tubing,
+                "z_drop": z_drop,
+                "viscosity": viscosity,
+            }
+        )
+        if device not in ["x_mantis", "x_tempest_chip"]:
+            raise ValueError(
+                f"Device is {device}. It must be: [x_tempest_chip, x_mantis]"
+            )
+        if any(params.values()):
+            LiquidHandleBuilders.validate_device_params(device, params)
+
+        device_dict: dict = {}
+        for key, value in params.items():
+            if value or isinstance(value, int):
+                if key == "nozzle_size" or key == "z_drop":
+                    device_dict.update({key: Unit(value)})
+                else:
+                    device_dict.update({key: value})
+
+        return {device: device_dict}
+
+    @staticmethod
+    def validate_device_params(device: str, device_dict: dict) -> None:
+        """
+        Helper validation function to validate device liquid handling params
+        Parameters
+        ----------
+        device: str
+            either x_mantis or x_tempest_chip
+        device_dict: dict
+            Dictionary of all device params, as seen in function device_mode_params
+
+        Raises
+        ------
+        ValueError
+            If input device is not either x_mantis or x_tempest_chip
+        ValueError
+            If input device_dict values are not of the accepted
+            params for the input device
+        KeyError
+            If input device_dict key is not in accepted_params keys
+        """
+        if device == "x_mantis":
+            # If device is mantis, set accepted params
+            accepted_params: dict = {
+                "model": ["high_volume", "low_volume"],
+                "diaphragm": [0, 100],
+                "nozzle_size": [Unit("0.1:mm"), Unit("0.2:mm"), Unit("0.5:mm")],
```

Comment:
butbutbutbut what if the user specifies `nozzle_size` as string?!

i think what we should do, before comparing to this dict, is call `parse_unit(NOZZLE_SIZE)` on the user-specified nozzle_size... and `z_drop`